src/lib/arch/riscv/Instruction_execute.cc

```diff
@@ -827,6 +841,642 @@ void Instruction::execute() {
+      const double rs2 = operands[1].get<double>();
+
+      const double xorSign = pow(-1, std::signbit(rs1) ^ std::signbit(rs2));
+
+      results[0] = RegisterValue(std::copysign(rs1, xorSign), 8);
+      break;
+    }
+    case Opcode::RISCV_FSGNJX_S: {  // FSGNJX.S rd,rs1,rs2
+      const float rs1 = operands[0].get<float>();
+      const float rs2 = operands[1].get<float>();
+
+      const float xorSign = pow(-1, std::signbit(rs1) ^ std::signbit(rs2));
+
+      results[0] = RegisterValue(NanBoxFloat(std::copysign(rs1, xorSign)), 8);
+      break;
+    }
+
+    case Opcode::RISCV_FMV_D_X: {  // FMV.D.X rd,rs1
+      const double rs1 = operands[0].get<double>();
+
+      results[0] = RegisterValue(rs1, 8);
+      break;
+    }
+    case Opcode::RISCV_FMV_X_D: {  // FMV.X.D rd,rs1
+      const double rs1 = operands[0].get<double>();
+
+      results[0] = RegisterValue(rs1, 8);
+      break;
+    }
+    case Opcode::RISCV_FMV_W_X: {  // FMV.W.X rd,rs1
+      const float rs1 = operands[0].get<float>();
+
+      results[0] = RegisterValue(NanBoxFloat(rs1), 8);
+      break;
+    }
+    case Opcode::RISCV_FMV_X_W: {  // FMV.X.W rd,rs1
+      const uint64_t rs1 = operands[0].get<uint64_t>();
+
+      results[0] = RegisterValue(signExtendW(rs1), 8);
+      break;
+    }
+
+      // TODO FLT.S and FLE.S perform what the IEEE 754-2008 standard refers to
+      // as signaling comparisons: that is, they set the invalid operation
+      // exception flag if either input is NaN. FEQ.S performs a quiet
+      // comparison: it only sets the invalid operation exception flag if either
+      // input is a signaling NaN. For all three instructions, the result is 0
+      // if either operand is NaN.
+    case Opcode::RISCV_FEQ_D: {  // FEQ.D rd,rs1,rs2
+      // TODO FEQ.S performs a quiet
+      // comparison: it only sets the invalid operation exception flag if
+      // either input is a signaling NaN. Qemu doesn't seem to set CSR flags
+      // with sNANs
+      const double rs1 = operands[0].get<double>();
+      const double rs2 = operands[1].get<double>();
+
+      if (rs1 == rs2 && !std::isnan(rs1) && !std::isnan(rs2)) {
+        results[0] = RegisterValue(static_cast<uint64_t>(1), 8);
+      } else {
+        results[0] = RegisterValue(static_cast<uint64_t>(0), 8);
+      }
+      break;
+    }
+    case Opcode::RISCV_FEQ_S: {  // FEQ.S rd,rs1,rs2
+      const float rs1 = operands[0].get<float>();
+      const float rs2 = operands[1].get<float>();
+
+      if (rs1 == rs2 && !std::isnan(rs1) && !std::isnan(rs2)) {
+        results[0] = RegisterValue(static_cast<uint64_t>(1), 8);
+      } else {
+        results[0] = RegisterValue(static_cast<uint64_t>(0), 8);
+      }
+      break;
+    }
+    case Opcode::RISCV_FLT_D: {  // FLT.D rd,rs1,rs2
+      const double rs1 = operands[0].get<double>();
+      const double rs2 = operands[1].get<double>();
+
+      if (std::isnan(rs1) || std::isnan(rs2)) {
+        // TODO set csr flag
```

Comment:
Is there something blocking this TODO and all similar csr TODOs?

Reply:
The control and status register implementation is not complete/non-existent and so does not support updating CSR values

Reply:
I suggest I look at the new Huawei CSR implementation as this seems more complete. I can convert this into a PR later but leave CSR instructions with dummy implementations for now as they are not integral to the codes that we are currently running but some implementation is required for progression

Reply:
Could you change all these comments "TODO: set csr flag when Zicsr implementation is complete" to add clarity?

Reply:
Somehow missed on find and replace. Should be done now